app/webpacker/controllers/column_preferences_controller.js

```diff
@@ -30,14 +32,39 @@ export default class ColumnPreferencesController extends Controller {
     this.table.classList.toggle(`hide-${name}`, !element.checked);
 
     // Reset cell colspans
-    const hiddenColCount = this.checkboxes.filter((checkbox)=> !checkbox.checked).length;
```

Comment:
Before, this count was executed only once before looping through all colspan cells. Now it's running every time, which is less efficient. 
But.. I think it's not too expensive and is ok. I like how it's now shared with the observer.